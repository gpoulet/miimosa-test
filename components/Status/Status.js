import { string } from 'prop-types'
import Chip from '@material-ui/core/Chip'
import { getColorFromStatus, getLabelFromStatus } from './Status.util'

function Status ({ status }) {
  return (
    <Chip className="statusChip" label={getLabelFromStatus(status)} color={getColorFromStatus(status)} />
  )
}

Status.propTypes = {
  status: string.isRequired
}

export default Status