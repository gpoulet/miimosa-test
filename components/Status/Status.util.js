export const getColorFromStatus = status => {
  switch (status) {
    case "success": return "primary"
    case "in_progress": return "secondary"
    default : return "default"
  }
}

export const getLabelFromStatus = status => {
  switch (status) {
    case "success": return "SuccÃ¨s"
    case "in_progress": return "En cours"
    case "failed": return "Echec"
    default : return ""
  }
}