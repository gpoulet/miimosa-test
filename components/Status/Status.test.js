import React from 'react';
import renderer from 'react-test-renderer';
import Status from './Status';

it('Status success renders correctly', () => {
  const tree = renderer
    .create(<Status status="success" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Status in_progress renders correctly', () => {
  const tree = renderer
    .create(<Status status="in_progress" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Status failed renders correctly', () => {
  const tree = renderer
    .create(<Status status="failed" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});