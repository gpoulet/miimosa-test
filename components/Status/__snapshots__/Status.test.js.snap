// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status failed renders correctly 1`] = `
<div
  className="MuiChip-root statusChip"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label"
  >
    Echec
  </span>
</div>
`;

exports[`Status in_progress renders correctly 1`] = `
<div
  className="MuiChip-root statusChip MuiChip-colorSecondary"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label"
  >
    En cours
  </span>
</div>
`;

exports[`Status success renders correctly 1`] = `
<div
  className="MuiChip-root statusChip MuiChip-colorPrimary"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label"
  >
    Succ√®s
  </span>
</div>
`;
